### ISSUE TITLE
Source Generated interop should warn when a user puts `[OutAttribute]` on a `string` and suggest `[Out] char[]`

### ISSUE THREAD

**Issue by jtschuster** [dotnet/runtime#88617]
LibraryImportGenerator and ComInterfaceGenerator allow `[Out]` on a `string`, which implies the native method will modify the string, which can lead to runtime corruption. The interop generators should warn if they see this and suggest a `char` array or a `ref`/`out` parameter instead.

**Comment by AaronRobinsonMSFT** [comment]
>  The interop generators should warn if they see this

I think we might just want an error. I can't recall if this is one of those things that warning/error are the same thing, but I would argue this should be a hard fail as it is wrong.

**Comment by jkoritzinsky** [comment]
I agree with @AaronRobinsonMSFT here. This should be an error as it can lead to runtime corruption. It's only a warning for P/Invokes because we couldn't make it an error.
